test/hts-precompile/token-create/tokenCreateContract.js

```diff
@@ -91,4 +90,72 @@ describe("TokenCreateContract tests", function () {
+    const tokenAddressTx = await tokenCreateContract.createFungibleTokenPublic(tokenCreateContract.address, {
+      value: ethers.BigNumber.from('10000000000000000000'),
+      gasLimit: 1_000_000
+    });
+    const tokenAddressReceipt = await tokenAddressTx.wait();
+    const result = tokenAddressReceipt.events.filter(e => e.event === 'CreatedToken')[0].args[0];
+    expect(result).to.exist;
+    expectValidHash(result, 40)
+  });
+
+  it('should be able to execute createNonFungibleToken', async function () {
+    const tokenAddressTx = await tokenCreateContract.createNonFungibleTokenPublic(tokenCreateContract.address, {
+      value: ethers.BigNumber.from('10000000000000000000'),
+      gasLimit: 1_000_000
+    });
+
+    const tokenAddressReceipt = await tokenAddressTx.wait();
+    const result = tokenAddressReceipt.events.filter(e => e.event === 'CreatedToken')[0].args[0];
+    expect(result).to.exist;
+    expectValidHash(result, 40)
+  });
+
+  it('should be able to execute createFungibleTokenWithCustomFees', async function () {
+    const tx = await tokenCreateContract.createFungibleTokenWithCustomFeesPublic(signers[0].address, tokenAddress, {
+      value: ethers.BigNumber.from('20000000000000000000'),
+      gasLimit: 1_000_000
+    });
+
+    const txReceipt = await tx.wait();
+    const result = txReceipt.events.filter(e => e.event === 'CreatedToken')[0].args[0];
+    expect(result).to.exist;
+    expectValidHash(result, 40)
+  });
+
+  it('should be able to execute createNonFungibleTokenWithCustomFees', async function () {
+    const tx = await tokenCreateContract.createNonFungibleTokenWithCustomFeesPublic(signers[0].address, tokenAddress, {
+      value: ethers.BigNumber.from('20000000000000000000'),
+      gasLimit: 1_000_000
+    });
+
+    const txReceipt = await tx.wait();
+    const result = txReceipt.events.filter(e => e.event === 'CreatedToken')[0].args[0];
+    expect(result).to.exist;
+    expectValidHash(result, 40)
+  });
+
+  it('should be able to execute mintToken', async function () {
+    const nftAddress = await utils.createNonFungibleToken(tokenCreateContract);
+    expect(nftAddress).to.exist;
+    expectValidHash(nftAddress, 40);
+
+    const tx = await tokenCreateContract.mintTokenPublic(nftAddress, 0, ['0x02'], {
+      gasLimit: 1_000_000
+    });
+
+    const receipt = await tx.wait();
+    const { responseCode } = receipt.events.filter(e => e.event === 'ResponseCode')[0].args;
+    expect(responseCode).to.equal(22);
+    const { serialNumbers } = receipt.events.filter(e => e.event === 'MintedToken')[0].args;
+    expect(serialNumbers[0].toNumber()).to.be.greaterThan(0);
+  });
+
+  it('should be able to execute grantTokenKyc', async function () {
+    const grantKycTx = await tokenCreateContract.grantTokenKycPublic(tokenAddress, signers[1].address, { gasLimit: 1000000 });
+    expect((await grantKycTx.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.equal(22);
+  });
```

Comment:
Could you add `unfreeze()` and `unpause()` from the contract

Reply:
I moved those methods to `TokenManagementContract` since they and `freeze()` and `pause()` are all part of the same HIP. It didn't really make sence for those methods to be separated in different contracts.

Reply:
Sure, the original intention noted in the issues was
token create - any actions does to create and enable token transfers
token transfer - transfer operations
token manage - token and token account modifications beyond enablement
token query - token info retrieval

The `unfreeze`, `unpause` and `kycGranted` are debatable so i'm fine with what you have.